test/widgets/action_sheet_test.dart

```diff
@@ -99,46 +107,244 @@ void main() {
-      await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+    group('emoji picker;', () {
+      Future<void> setupEmojiPicker(WidgetTester tester, {
+        required Message message,
+        required Narrow narrow,
+      }) async {
+        final httpClient = FakeImageHttpClient();
+        debugNetworkImageHttpClientProvider = () => httpClient;
+        httpClient.request.response
+          ..statusCode = HttpStatus.ok
+          ..content = kSolidBlueAvatar;
+
+        await setupToMessageActionSheet(tester, message: message, narrow: narrow);
+        store.setServerEmojiData(ServerEmojiData(codeToNames: {
+          '1f4a4': ['zzz', 'sleepy'], // (just 'zzz' in real data)
+        }));
+        await store.handleEvent(RealmEmojiUpdateEvent(id: 1, realmEmoji: {
+          '1': eg.realmEmojiItem(emojiCode: '1', emojiName: 'buzzing'),
+        }));
+
+        await tester.tap(find.ancestor(
+          of: find.byIcon(ZulipIcons.chevron_right),
+          matching: find.byType(InkWell)));
+        await tester.pumpAndSettle();
+        await tester.ensureVisible(find.byType(EmojiPicker));
+      }
 
-      connection.prepare(httpStatus: 400, json: {
-        'code': 'BAD_REQUEST',
-        'msg': 'Invalid message(s)',
-        'result': 'error',
+      final searchFieldFinder = find.widgetWithText(TextField, 'Search emoji');
+
+      Condition<Object?> conditionEmojiListEntry({
+        required ReactionType emojiType,
+        required String emojiCode,
+        required String emojiName,
+      }) {
+        return (Subject<Object?> it) => it.isA<EmojiPickerListEntry>()
+          ..emoji.which((it) => it
+            ..emojiType.equals(emojiType)
+            ..emojiCode.equals(emojiCode)
+            ..emojiName.equals(emojiName));
+      }
+
+      List<Condition<Object?>> arePopularCandidates = popularCandidates.map((c) =>
+        conditionEmojiListEntry(
+          emojiType: c.emojiType,
+          emojiCode: c.emojiCode,
+          emojiName: c.emojiName)).toList();
+
+      testWidgets('show, search', (tester) async {
+        final message = eg.streamMessage();
+        await setupEmojiPicker(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+
+        check(tester.widgetList<EmojiPickerListEntry>(find.byType(EmojiPickerListEntry))).deepEquals([
+          ...arePopularCandidates,
+          conditionEmojiListEntry(
+            emojiType: ReactionType.realmEmoji,
+            emojiCode: '1',
+            emojiName: 'buzzing'),
+          conditionEmojiListEntry(
+            emojiType: ReactionType.zulipExtraEmoji,
+            emojiCode: 'zulip',
+            emojiName: 'zulip'),
+          conditionEmojiListEntry(
+            emojiType: ReactionType.unicodeEmoji,
+            emojiCode: '1f4a4',
+            emojiName: 'zzz'),
+        ]);
+
+        tester.widget(searchFieldFinder);
+        await tester.enterText(searchFieldFinder, 'z');
```

Comment:
```suggestion
        await tester.enterText(searchFieldFinder, 'z');
```

Or is there an effect the `tester.widget` line has which I'm missing?